### PR TITLE
#2142. Change dynamic semantics member invocation tests to be extension type tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Inline-classes/dynamic_semantics_member_invocation_A02_t03.dart [hunk 1]
```diff
@@ -26,30 +28,24 @@
...
-inline class IC2<T> implements IC<T>{
-  final T id;
-  IC2(this.id);
-}
+extension type ET2<T> implements ET1<T>(T id) {}
 
 main() {
-  IC<int> ic_1 = IC(42);
-  var asMapTearOff_1 = ic_1.asMap;
+  ET1<int> ic_1 = ET1(42);
```
Comment (eernstg) [inline LanguageFeatures/Inline-classes/dynamic_semantics_member_invocation_A02_t03.dart:38]: Maybe:
```suggestion
  ET1<int> et1_1 = ET1(42);
```

File: LanguageFeatures/Inline-classes/dynamic_semantics_member_invocation_A03_t01.dart [hunk 2]
```diff
@@ -31,40 +34,30 @@
...
+  ET1<num> et2 = ET2(42);
   Expect.mapEquals(
-      {"key1": int, "key2": num}, ic2.asMap<String, int>("key1", "key2"));
+      {"key1": int, "key2": num}, et2.asMap<String, int>("key1", "key2"));
   Expect.mapEquals(
-      {"key1": num, "key2": num}, ic2.asMap<String, num>("key1", "key2"));
+      {"key1": num, "key2": num}, et2.asMap<String, num>("key1", "key2"));
 
-  IC3<int> ic3 = IC3(42);
+  ET3<int> ic3 = ET3(42);
```
Comment (eernstg) [inline LanguageFeatures/Inline-classes/dynamic_semantics_member_invocation_A03_t01.dart:58]:
```suggestion
  ET3<int> et3 = ET3(42);
```

File: LanguageFeatures/Inline-classes/dynamic_semantics_member_invocation_A02_t03.dart [hunk 3]
```diff
@@ -26,30 +28,24 @@
...
-  IC<num> ic_2 = IC(42);
-  var asMapTearOff_2 = ic_2.asMap;
+  ET1<num> et1_2 = ET1(42);
+  var asMapTearOff_2 = et1_2.asMap;
   Expect.notEquals(asMapTearOff_2, asMapTearOff_1);
   Expect.equals(asMapTearOff_2, asMapTearOff_2);
 
-  IC2<num> ic2 = IC2(42);
-  var asMapTearOff2 = ic2.asMap;
+  IC2<num> et2 = IC2(42);
```
Comment (eernstg) [inline LanguageFeatures/Inline-classes/dynamic_semantics_member_invocation_A02_t03.dart:47]: Maybe:
```suggestion
  ET2<num> et2 = ET2(42);
```

**Review by eernstg — COMMENTED** [review]
Looks good! I noticed a couple of typos. 

By the way, checking the branch locally, I noticed one similar typo in a file which isn't included: syntax_A09_t01.dart has `inline class IC3` but uses `ET3`.

**Comment by sgrekhov** [comment]
Typos fixed.

> By the way, checking the branch locally, I noticed one similar typo in a file which isn't included: syntax_A09_t01.dart has inline class IC3 but uses ET3
Fixed by merging with `master`. Please take another look

**Pull request merged**